src/core/or/policies.c

```diff
@@ -1048,6 +1053,12 @@ fascist_firewall_choose_address_ls(const smartlist_t *lspecs,
     }
   } SMARTLIST_FOREACH_END(ls);
 
+  /* If we don't have IPv4 or IPv6 in link specifiers, log a bug and return. */
+  if (!have_v4 && !have_v6) {
+    log_warn(LD_BUG, "None of our link specifiers have IPv4 or IPv6");
```

Comment:
IPv4 is required. So a missing IPv4 in the link specifiers isn't a bug in this tor: it's a problem with the data that was sent to this tor.
Let's make it a protocol warning.

If IPv6 is also missing, we should change the protocol warning message.